src/main/java/org/sciborgs1155/robot/drive/Drive.java

```diff
@@ -1,263 +1,181 @@
-        : new Drive(new SimModule(), new SimModule(), new SimModule(), new SimModule());
-  }
+  /**
+   * Encoders PID and FF controllers, -> mostly done by now, maybe there is something else about this that can be worked on later
+   * DifferentialDriveOdometry -> used DifferentialDrivePoseEstimator instead rn
+   * Getters for pose, encoder values
+   * Setters for setpoints and other important values
+   * Style
+   */
+  
 
-  public Drive(ModuleIO frontLeft, ModuleIO frontRight, ModuleIO rearLeft, ModuleIO rearRight) {
-    this.frontLeft = frontLeft;
-    this.frontRight = frontRight;
-    this.rearLeft = rearLeft;
-    this.rearRight = rearRight;
+  //sets up Odometry
+  private final DifferentialDriveKinematics kinematics = new DifferentialDriveKinematics(ROBOT_TRACK);
+  DifferentialDriveWheelSpeeds speeds = new DifferentialDriveWheelSpeeds(0, 0);
 
-    modules = List.of(frontLeft, frontRight, rearLeft, rearRight);
-    modules2d = new FieldObject2d[modules.size()];
+  @Log private final DifferentialDrivePoseEstimator odometry = 
+    new DifferentialDrivePoseEstimator(
+      kinematics,
+      gyro.getRotation2d(), 
+      leftEncoder.getPosition(), 
+      rightEncoder.getPosition(), 
+      new Pose2d(),
+          VecBuilder.fill(0.05, 0.05, Units.degreesToRadians(5)), //these could probably be adjusted later, taken from wpilib docs
+          VecBuilder.fill(0.5, 0.5, Units.degreesToRadians(30)));
 
-    odometry =
-        new SwerveDrivePoseEstimator(kinematics, getHeading(), getModulePositions(), new Pose2d());
+  /** Creates a new Drive. */
+  public Drive(){
+    leftEncoder.setPositionConversionFactor(DriveConstants.GEAR_RATIO * DriveConstants.WHEEL_CIRCUMFERENCE);
+    rightEncoder.setPositionConversionFactor(DriveConstants.GEAR_RATIO * DriveConstants.WHEEL_CIRCUMFERENCE);
 
-    for (int i = 0; i < modules2d.length; i++) {
-      modules2d[i] = field2d.getObject("module-" + i);
-    }
-  }
+    leftEncoder.setVelocityConversionFactor(DriveConstants.GEAR_RATIO);
+    leftEncoder.setVelocityConversionFactor(DriveConstants.GEAR_RATIO);
 
-  /**
-   * Returns the currently-estimated pose of the robot.
-   *
-   * @return The pose.
-   */
-  public Pose2d getPose() {
-    return odometry.getEstimatedPosition();
-  }
+    rightMotors.setInverted(true);
+    
+    // AutoBuilder.configureLTV()
 
-  /**
-   * Returns the heading of the robot, based on our pigeon
-   *
-   * @return A Rotation2d of our angle
-   */
-  public Rotation2d getHeading() {
-    return imu.getRotation2d();
+    
   }
 
-  /**
-   * Resets the odometry to the specified pose.
-   *
-   * @param pose The pose to which to set the odometry.
-   */
-  public void resetOdometry(Pose2d pose) {
-    odometry.resetPosition(getHeading(), getModulePositions(), pose);
+  public void setVoltage(double voltageR, double voltageL) {
+    rightMotors.setVoltage(MathUtil.clamp(voltageR, -MAX_SPEED, MAX_SPEED)); 
```

Comment:
You shouldn't clamp voltage in between min and max speeds, since speed is a velocity value, measured in m/s, and voltage is measured in volts